proposals/10-egress-gateways.md

```diff
@@ -88,19 +88,119 @@ ingress use case.
-> in other projects. However, the general guidance is to break things down into
-> highly focused sections as much as possible to help make things easier to
-> read and review. Long, unbroken walls of code and YAML in this document are
-> not advisable as that may increase the time it takes to review.
+1. Resource model using Gateway + HTTPRoute with a Backend for destinations (Service or FQDN).
+2. Two routing modes: Endpoint (direct) and Parent (gateway chaining).
+3. Policy scoping: Gateway (global posture), Route (filters, per-request), Backend (per-destination).
+4. Extension points for AI use cases (payload processing, guardrails), without assuming an AI-only design.
+
+
+## Open Design Questions
+
+### Gateway Resource: Reuse vs. New Type
+
+Two approaches are under consideration:
+
+**Option A: Reuse Gateway API Gateway**
+- Leverage existing `Gateway`, `HTTPRoute`, and `GRPCRoute` resources
+- HTTPRoute references to external backends make it an egress gateway
+- Requires Backend resource to represent external destinations
+
+**Option B: New EgressGateway Resource**
+- Introduce dedicated `EgressGateway` resource type
+- Enables egress-specific fields (e.g., global CIDR allow-lists) without policy attachment overhead
+- Clearer separation of ingress vs egress concerns
+
+Option B implies defining equivalents of parentRefs, listeners, and route attachment; this is a significant fork from Gateway API and should be justified by clear need for an egress specific spec.
+
+**Recommendation needed**: Feedback requested on whether the semantics justify a new resource or if Gateway reuse is sufficient.
+
+### Backend Resource and Policy Application
+
+To avoid heavy reliance on policy attachment, embed **typed** policy into `Backend` with clear phases and failure semantics, or use separate CRDs with `targetRef: Backend`.
+
+```yaml
+apiVersion: agentic.networking.x-k8s.io/v1alpha1
+kind: Backend
+metadata:
+  name: openai-backend
+spec:
+  destination:
+    type: FQDN
+    fqdn:
+      hostname: api.openai.com
+      port: 443
+  tls:
+    mode: SIMPLE            # SIMPLE | MUTUAL
+    serverName: api.openai.com
+    caBundleRef:
+      name: vendor-ca
+    # clientCertificateRef:  # if MUTUAL
+    #   name: egress-client-cert
+  extensions:
+  - name: inject-credentials
+    type: CredentialInjector
+    phase: request-headers   # request-headers|request-body|connect|backend-request|backend-response|response-body|response-headers
+    priority: 10
+    failOpen: false
+    config:
+      secretRef:
+        name: openai-api-key
+        namespace: platform-secrets
+  - name: rate-qos
+    type: QoSController
+    phase: backend-request
+    priority: 30
+    failOpen: true
+    config:
+      requestsPerMinute: 1000
+```
+Alternatively, policies MAY be separate CRDs (e.g., `BackendTLSPolicy`, `EgressPolicy`) with `spec.targetRef: Backend`, avoiding schema growth on `Backend`.
```

Comment:
I definitely see the schema growth issue...does something like Envoy's typed config model help out a bit? Give implementations a way to dynamically instantiate a policy based on well-known type (urls?)?